compiler/bindings/python/test/api/api_test.py

```diff
@@ -81,6 +81,22 @@ def testInputBuffer(self):
             self.assertIn(b"module", bytes(mem))
             out.close()
 
+        def testBytecodeInputBuffer(self):
+            session = Session()
+            inv = session.invocation()
+            source = Source.wrap_buffer(
+                session,
+                bytes.fromhex(
+                    "4d4cef520d4d4c495231392e302e3067697400010d030103010307030d05010105130b020d170303030505041705015001010704070301010603010501004707210f116275696c74696e006d6f64756c65002e2e2f2e2e2f746573742e6d6c697200080903050101",
+                ),
```

Comment:
At the bare minimum this would need a comment documenting how the hex was generated.
> The bytecode in the test is just builtin.module {}

Just including that outside of the code in a PR discussion comment is not sufficient.

However, instead I think this should either be left out of the PR entirely (try to keep the PR focused on one issue, this is scope creeping) or the hex should be generated as part of the test (e.g. by writing bytecode and then trying to read it, roundtrip style).

Reply:
+1 on this, I will revert the last commit. Let's merge this PR it only __main__ is bothering and I will open a follow up PR that fixes this bug and removes the dependency